transit_odp/organisation/csv/overall.py

```diff
@@ -89,6 +95,54 @@
     "string_lines",
 )
 
+DATACATALOGUE_ATTRIBUTE_FIELDS = (
+    "fares_metadata_id",
+    "xml_file_name",
+    "national_operator_code",
+    "line_name",
+)
+
+
+def get_fares_data():
+    """
+    Returns list of fares information (xml file name, NOCs and line name)
+    when dataset_id in DatasetRevision table match
+    fares_metadata_id in DataCatalogueMetaData table.
+    """
+    dataset_ids_dict = (
+        DataCatalogueMetaData.objects.filter(
+            fares_metadata_id__in=FaresMetadata.objects.all().values_list(
+                "datasetmetadata_ptr"
+            )
+        )
+        .order_by("fares_metadata_id")
+        .values(*DATACATALOGUE_ATTRIBUTE_FIELDS)
+    )
+
+    for dataset_id in dataset_ids_dict:
+        revision_ids = DatasetMetadata.objects.filter(
+            id=dataset_id.get("fares_metadata_id")
+        ).values("revision_id")
+        for revision_id in revision_ids:
+            dataset_id["revision_id"] = revision_id["revision_id"]
+
+        dataset_ids = DatasetRevision.objects.filter(
+            id=dataset_id.get("revision_id")
+        ).values("dataset_id")
+        for dataset_id in dataset_ids:
+            dataset_id["dataset_id"] = dataset_id["dataset_id"]
+
+    return list(dataset_ids_dict)
+
+
+def transform_lists(data_list):
```

Comment:
This can be done using a stingagg on the database, You will see other examples of how to do this in the queryset modules